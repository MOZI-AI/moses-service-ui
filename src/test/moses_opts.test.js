import React from 'react'
import { mount } from 'enzyme'
import { MosesOptionsForm } from '../moses_opts'
import renderer from 'react-test-renderer';

describe('<MosesOptionsForm />', () => {

    it('rendered correctly', () => {
        const tree = renderer.create(<MosesOptionsForm show='true' />).toJSON()
        expect(tree).toMatchSnapshot()
    });


    it('shows error when attempting to add duplicate additional parameter', () => {
        const wrapper = mount(<MosesOptionsForm additionalParameters={{ a: 1 }} show='true' />)
        const name = wrapper.find('Input[name="name"]').first()

        name.simulate('change', { target: { value: 'a' } })
        expect(wrapper.find('.has-error input[name="name"]').exists()).toBeTruthy()
    })

    it('hides feature selection options when feature selection is disabled', () => {
        const wrapper = mount(<MosesOptionsForm defaults={{ enableFeatureSelection: false }} show='true' />)
        expect(wrapper.find('div#featureSelectionAlgorithm').exists()).toBeFalsy()
        expect(wrapper.find('Input[name="featureSelectionTargetSize"]').exists()).toBeFalsy()
    })

    it('hides hcWidenSearch options when hcWidenSearch is off', () => {
        const wrapper = mount(<MosesOptionsForm defaults={{ hcWidenSearch: false }} show='true' />)
        expect(wrapper.find('Input[name="hcCrossoverMinNeighbors"]').exists()).toBeFalsy()
        expect(wrapper.find('Input[name="hcCrossoverPopSize"]').exists()).toBeFalsy()
    })

})