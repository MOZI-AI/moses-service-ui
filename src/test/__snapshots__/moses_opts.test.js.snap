// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MosesOptionsForm /> rendered correctly 1`] = `
<div
  className="mosesOptionsFormWrapper"
>
  <form>
    <div
      className="MuiGrid-container-1 MuiGrid-spacing-xs-16-24"
    >
      <div
        className="MuiGrid-item-2 MuiGrid-grid-xs-12-41 MuiGrid-grid-sm-6-49 MuiGrid-grid-md-4-61"
      >
        <label
          aria-describedby={null}
          className="MuiFormControlLabel-root-106"
          onBlur={[Function]}
          onFocus={[Function]}
          onMouseLeave={[Function]}
          onMouseOver={[Function]}
          onTouchEnd={[Function]}
          onTouchStart={[Function]}
          title="Enable integrated feature selection.  Feature selection is performed immediately before knob building (representation building), when creating a new deme."
        >
          <span
            className="MuiSwitch-root-112"
          >
            <span
              className="MuiButtonBase-root-131 MuiIconButton-root-125 MuiPrivateSwitchBase-root-121 MuiSwitch-switchBase-115 MuiSwitch-colorSecondary-118"
              onBlur={[Function]}
              onContextMenu={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex={null}
            >
              <span
                className="MuiIconButton-label-130"
              >
                <span
                  className="MuiSwitch-icon-113"
                />
                <input
                  className="MuiPrivateSwitchBase-input-124"
                  defaultChecked={false}
                  name="enableFeatureSelection"
                  onChange={[Function]}
                  type="checkbox"
                  value="0"
                />
              </span>
              <span
                className="MuiTouchRipple-root-266"
              />
            </span>
            <span
              className="MuiSwitch-bar-120"
            />
          </span>
          <span
            className="MuiTypography-root-134 MuiTypography-body1-143 MuiFormControlLabel-label-111"
          >
            Enable feature selection
          </span>
        </label>
      </div>
      <div
        className="MuiGrid-item-2 MuiGrid-grid-xs-12-41 MuiGrid-grid-sm-6-49 MuiGrid-grid-md-4-61"
      >
        <label
          aria-describedby={null}
          className="MuiFormControlLabel-root-106"
          onBlur={[Function]}
          onFocus={[Function]}
          onMouseLeave={[Function]}
          onMouseOver={[Function]}
          onTouchEnd={[Function]}
          onTouchStart={[Function]}
          title="Hillclimbing parameter (hc). If false,then deme search terminates when a local hilltop is found. If true, then the search radius is progressively widened, until another termination condition is met. "
        >
          <span
            className="MuiSwitch-root-112"
          >
            <span
              className="MuiButtonBase-root-131 MuiIconButton-root-125 MuiPrivateSwitchBase-root-121 MuiSwitch-switchBase-115 MuiSwitch-colorSecondary-118"
              onBlur={[Function]}
              onContextMenu={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex={null}
            >
              <span
                className="MuiIconButton-label-130"
              >
                <span
                  className="MuiSwitch-icon-113"
                />
                <input
                  className="MuiPrivateSwitchBase-input-124"
                  defaultChecked={false}
                  name="hcWidenSearch"
                  onChange={[Function]}
                  type="checkbox"
                  value="0"
                />
              </span>
              <span
                className="MuiTouchRipple-root-266"
              />
            </span>
            <span
              className="MuiSwitch-bar-120"
            />
          </span>
          <span
            className="MuiTypography-root-134 MuiTypography-body1-143 MuiFormControlLabel-label-111"
          >
            hc Widen Search
          </span>
        </label>
      </div>
      <div
        className="MuiGrid-item-2 MuiGrid-grid-xs-12-41 MuiGrid-grid-sm-6-49 MuiGrid-grid-md-4-61"
      >
        <label
          aria-describedby={null}
          className="MuiFormControlLabel-root-106"
          onBlur={[Function]}
          onFocus={[Function]}
          onMouseLeave={[Function]}
          onMouseOver={[Function]}
          onTouchEnd={[Function]}
          onTouchStart={[Function]}
          title="If the table has discrete output type (like bool or enum), balance the resulting ctable so all classes have the same weight."
        >
          <span
            className="MuiSwitch-root-112"
          >
            <span
              className="MuiButtonBase-root-131 MuiIconButton-root-125 MuiPrivateSwitchBase-root-121 MuiSwitch-switchBase-115 MuiSwitch-colorSecondary-118 MuiPrivateSwitchBase-checked-122 MuiSwitch-checked-116"
              onBlur={[Function]}
              onContextMenu={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex={null}
            >
              <span
                className="MuiIconButton-label-130"
              >
                <span
                  className="MuiSwitch-icon-113 MuiSwitch-iconChecked-114"
                />
                <input
                  className="MuiPrivateSwitchBase-input-124"
                  defaultChecked={true}
                  name="balance"
                  onChange={[Function]}
                  type="checkbox"
                  value="0"
                />
              </span>
              <span
                className="MuiTouchRipple-root-266"
              />
            </span>
            <span
              className="MuiSwitch-bar-120"
            />
          </span>
          <span
            className="MuiTypography-root-134 MuiTypography-body1-143 MuiFormControlLabel-label-111"
          >
            Balance
          </span>
        </label>
      </div>
    </div>
    <div
      className="MuiGrid-container-1 MuiGrid-spacing-xs-24-25"
    >
      <div
        className="MuiGrid-item-2 MuiGrid-grid-xs-12-41 MuiGrid-grid-sm-6-49 MuiGrid-grid-md-4-61"
      >
        <div
          aria-describedby={null}
          className="MuiFormControl-root-170 MuiFormControl-marginDense-172 MuiFormControl-fullWidth-173"
          onMouseLeave={[Function]}
          onMouseOver={[Function]}
          onTouchEnd={[Function]}
          onTouchStart={[Function]}
          title="Maximum number of fitness function evaluations."
        >
          <label
            className="MuiFormLabel-root-185 MuiInputLabel-root-174 MuiInputLabel-formControl-179 MuiInputLabel-animated-182 MuiInputLabel-marginDense-180 MuiInputLabel-outlined-184"
            data-shrink={false}
          >
            Maximum evals
          </label>
          <div
            className="MuiInputBase-root-205 MuiOutlinedInput-root-192 MuiInputBase-fullWidth-214 MuiInputBase-formControl-206 MuiInputBase-marginDense-212"
            onClick={[Function]}
          >
            <fieldset
              aria-hidden={true}
              className="MuiPrivateNotchedOutline-root-222 MuiOutlinedInput-notchedOutline-199"
              style={
                Object {
                  "paddingLeft": 8,
                }
              }
            >
              <legend
                className="MuiPrivateNotchedOutline-legend-223"
                style={
                  Object {
                    "width": 0.01,
                  }
                }
              >
                <span
                  dangerouslySetInnerHTML={
                    Object {
                      "__html": "&#8203;",
                    }
                  }
                />
              </legend>
            </fieldset>
            <input
              aria-invalid={false}
              className="MuiInputBase-input-215 MuiOutlinedInput-input-200 MuiInputBase-inputMarginDense-216 MuiOutlinedInput-inputMarginDense-201"
              defaultValue={1000}
              disabled={false}
              name="maximumEvals"
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              placeholder="Maximum evals"
              required={false}
              type="text"
            />
          </div>
        </div>
      </div>
      <div
        className="MuiGrid-item-2 MuiGrid-grid-xs-12-41 MuiGrid-grid-sm-6-49 MuiGrid-grid-md-4-61"
      >
        <div
          aria-describedby={null}
          className="MuiFormControl-root-170 MuiFormControl-marginDense-172 MuiFormControl-fullWidth-173"
          onMouseLeave={[Function]}
          onMouseOver={[Function]}
          onTouchEnd={[Function]}
          onTouchStart={[Function]}
          title="The number of results to return,ordered according to a linear combination of score and complexity. If negative, then return all results."
        >
          <label
            className="MuiFormLabel-root-185 MuiInputLabel-root-174 MuiInputLabel-formControl-179 MuiInputLabel-animated-182 MuiInputLabel-marginDense-180 MuiInputLabel-outlined-184"
            data-shrink={false}
          >
            Result Count
          </label>
          <div
            className="MuiInputBase-root-205 MuiOutlinedInput-root-192 MuiInputBase-fullWidth-214 MuiInputBase-formControl-206 MuiInputBase-marginDense-212"
            onClick={[Function]}
          >
            <fieldset
              aria-hidden={true}
              className="MuiPrivateNotchedOutline-root-222 MuiOutlinedInput-notchedOutline-199"
              style={
                Object {
                  "paddingLeft": 8,
                }
              }
            >
              <legend
                className="MuiPrivateNotchedOutline-legend-223"
                style={
                  Object {
                    "width": 0.01,
                  }
                }
              >
                <span
                  dangerouslySetInnerHTML={
                    Object {
                      "__html": "&#8203;",
                    }
                  }
                />
              </legend>
            </fieldset>
            <input
              aria-invalid={false}
              className="MuiInputBase-input-215 MuiOutlinedInput-input-200 MuiInputBase-inputMarginDense-216 MuiOutlinedInput-inputMarginDense-201"
              defaultValue={100}
              disabled={false}
              name="resultCount"
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              placeholder="Result Count"
              required={false}
              type="text"
            />
          </div>
        </div>
      </div>
      <div
        className="MuiGrid-item-2 MuiGrid-grid-xs-12-41 MuiGrid-grid-sm-6-49 MuiGrid-grid-md-4-61"
      >
        <div
          aria-describedby={null}
          className="MuiFormControl-root-170 MuiFormControl-marginDense-172 MuiFormControl-fullWidth-173"
          onMouseLeave={[Function]}
          onMouseOver={[Function]}
          onTouchEnd={[Function]}
          onTouchStart={[Function]}
          title="Number of jobs allocated for deme optimization."
        >
          <label
            className="MuiFormLabel-root-185 MuiInputLabel-root-174 MuiInputLabel-formControl-179 MuiInputLabel-animated-182 MuiInputLabel-marginDense-180 MuiInputLabel-outlined-184"
            data-shrink={false}
          >
            Number Of Threads
          </label>
          <div
            className="MuiInputBase-root-205 MuiOutlinedInput-root-192 MuiInputBase-fullWidth-214 MuiInputBase-formControl-206 MuiInputBase-marginDense-212"
            onClick={[Function]}
          >
            <fieldset
              aria-hidden={true}
              className="MuiPrivateNotchedOutline-root-222 MuiOutlinedInput-notchedOutline-199"
              style={
                Object {
                  "paddingLeft": 8,
                }
              }
            >
              <legend
                className="MuiPrivateNotchedOutline-legend-223"
                style={
                  Object {
                    "width": 0.01,
                  }
                }
              >
                <span
                  dangerouslySetInnerHTML={
                    Object {
                      "__html": "&#8203;",
                    }
                  }
                />
              </legend>
            </fieldset>
            <input
              aria-invalid={false}
              className="MuiInputBase-input-215 MuiOutlinedInput-input-200 MuiInputBase-inputMarginDense-216 MuiOutlinedInput-inputMarginDense-201"
              defaultValue={8}
              disabled={false}
              name="numberOfThreads"
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              placeholder="Number Of Threads"
              required={false}
              type="text"
            />
          </div>
        </div>
      </div>
    </div>
    <div
      className="MuiGrid-container-1 MuiGrid-spacing-xs-24-25"
    >
      <div
        className="MuiGrid-item-2 MuiGrid-grid-xs-12-41 MuiGrid-grid-sm-6-49 MuiGrid-grid-md-4-61"
      >
        <legend
          aria-describedby={null}
          className="MuiFormLabel-root-185"
          onBlur={[Function]}
          onFocus={[Function]}
          onMouseLeave={[Function]}
          onMouseOver={[Function]}
          onTouchEnd={[Function]}
          onTouchStart={[Function]}
          title="Effort allocated for reduction during knob building, 0-3, 0 means minimum effort, 3 means maximum effort. Thebigger the effort the lower the dimension of the deme."
        >
          Reduct Knob Building Effort
        </legend>
        <div
          className="MuiFormGroup-root-224"
          defaultValue={0}
          role="radiogroup"
          style={
            Object {
              "display": "flex",
              "flexDirection": "row",
            }
          }
        >
          <label
            className="MuiFormControlLabel-root-106"
          >
            <span
              className="MuiButtonBase-root-131 MuiIconButton-root-125 MuiPrivateSwitchBase-root-121 MuiRadio-root-226 MuiRadio-colorSecondary-230"
              onBlur={[Function]}
              onContextMenu={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex={null}
            >
              <span
                className="MuiIconButton-label-130"
              >
                <svg
                  aria-hidden="true"
                  className="MuiSvgIcon-root-231"
                  focusable="false"
                  role="presentation"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
                  />
                </svg>
                <input
                  checked={false}
                  className="MuiPrivateSwitchBase-input-124"
                  name="reductKnobBuildingEffort"
                  onChange={[Function]}
                  type="radio"
                  value="0"
                />
              </span>
              <span
                className="MuiTouchRipple-root-266"
              />
            </span>
            <span
              className="MuiTypography-root-134 MuiTypography-body1-143 MuiFormControlLabel-label-111"
            >
              0
            </span>
          </label>
          <label
            className="MuiFormControlLabel-root-106"
          >
            <span
              className="MuiButtonBase-root-131 MuiIconButton-root-125 MuiPrivateSwitchBase-root-121 MuiRadio-root-226 MuiRadio-colorSecondary-230"
              onBlur={[Function]}
              onContextMenu={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex={null}
            >
              <span
                className="MuiIconButton-label-130"
              >
                <svg
                  aria-hidden="true"
                  className="MuiSvgIcon-root-231"
                  focusable="false"
                  role="presentation"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
                  />
                </svg>
                <input
                  checked={false}
                  className="MuiPrivateSwitchBase-input-124"
                  name="reductKnobBuildingEffort"
                  onChange={[Function]}
                  type="radio"
                  value="1"
                />
              </span>
              <span
                className="MuiTouchRipple-root-266"
              />
            </span>
            <span
              className="MuiTypography-root-134 MuiTypography-body1-143 MuiFormControlLabel-label-111"
            >
              1
            </span>
          </label>
          <label
            className="MuiFormControlLabel-root-106"
          >
            <span
              className="MuiButtonBase-root-131 MuiIconButton-root-125 MuiPrivateSwitchBase-root-121 MuiRadio-root-226 MuiRadio-colorSecondary-230"
              onBlur={[Function]}
              onContextMenu={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex={null}
            >
              <span
                className="MuiIconButton-label-130"
              >
                <svg
                  aria-hidden="true"
                  className="MuiSvgIcon-root-231"
                  focusable="false"
                  role="presentation"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
                  />
                </svg>
                <input
                  checked={false}
                  className="MuiPrivateSwitchBase-input-124"
                  name="reductKnobBuildingEffort"
                  onChange={[Function]}
                  type="radio"
                  value="2"
                />
              </span>
              <span
                className="MuiTouchRipple-root-266"
              />
            </span>
            <span
              className="MuiTypography-root-134 MuiTypography-body1-143 MuiFormControlLabel-label-111"
            >
              2
            </span>
          </label>
          <label
            className="MuiFormControlLabel-root-106"
          >
            <span
              className="MuiButtonBase-root-131 MuiIconButton-root-125 MuiPrivateSwitchBase-root-121 MuiRadio-root-226 MuiRadio-colorSecondary-230"
              onBlur={[Function]}
              onContextMenu={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex={null}
            >
              <span
                className="MuiIconButton-label-130"
              >
                <svg
                  aria-hidden="true"
                  className="MuiSvgIcon-root-231"
                  focusable="false"
                  role="presentation"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
                  />
                </svg>
                <input
                  checked={false}
                  className="MuiPrivateSwitchBase-input-124"
                  name="reductKnobBuildingEffort"
                  onChange={[Function]}
                  type="radio"
                  value="3"
                />
              </span>
              <span
                className="MuiTouchRipple-root-266"
              />
            </span>
            <span
              className="MuiTypography-root-134 MuiTypography-body1-143 MuiFormControlLabel-label-111"
            >
              3
            </span>
          </label>
        </div>
      </div>
      <div
        className="MuiGrid-item-2 MuiGrid-grid-xs-12-41 MuiGrid-grid-sm-6-49 MuiGrid-grid-md-4-61"
      >
        <div
          aria-describedby={null}
          className="MuiFormControl-root-170 MuiFormControl-marginDense-172 MuiFormControl-fullWidth-173"
          onMouseLeave={[Function]}
          onMouseOver={[Function]}
          onTouchEnd={[Function]}
          onTouchStart={[Function]}
          title="Fix the ratio of the score to complexity, to be used as a penalty, when ranking the metapopulation for fitness. ."
        >
          <label
            className="MuiFormLabel-root-185 MuiInputLabel-root-174 MuiInputLabel-formControl-179 MuiInputLabel-animated-182 MuiInputLabel-marginDense-180 MuiInputLabel-outlined-184"
            data-shrink={false}
          >
            Complexity ratio
          </label>
          <div
            className="MuiInputBase-root-205 MuiOutlinedInput-root-192 MuiInputBase-fullWidth-214 MuiInputBase-formControl-206 MuiInputBase-marginDense-212"
            onClick={[Function]}
          >
            <fieldset
              aria-hidden={true}
              className="MuiPrivateNotchedOutline-root-222 MuiOutlinedInput-notchedOutline-199"
              style={
                Object {
                  "paddingLeft": 8,
                }
              }
            >
              <legend
                className="MuiPrivateNotchedOutline-legend-223"
                style={
                  Object {
                    "width": 0.01,
                  }
                }
              >
                <span
                  dangerouslySetInnerHTML={
                    Object {
                      "__html": "&#8203;",
                    }
                  }
                />
              </legend>
            </fieldset>
            <input
              aria-invalid={false}
              className="MuiInputBase-input-215 MuiOutlinedInput-input-200 MuiInputBase-inputMarginDense-216 MuiOutlinedInput-inputMarginDense-201"
              defaultValue={3}
              disabled={false}
              name="complexityRatio"
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              placeholder="Complexity ratio"
              required={false}
              type="text"
            />
          </div>
        </div>
      </div>
    </div>
  </form>
  <form
    layout="inline"
    style={
      Object {
        "marginTop": "10px",
      }
    }
  >
    <div
      className="MuiGrid-container-1 MuiGrid-spacing-xs-24-25"
    >
      <div
        className="MuiGrid-item-2 MuiGrid-grid-xs-12-41"
      >
        <legend
          className="MuiFormLabel-root-185"
        >
          Additional parameters
        </legend>
        <div
          className="MuiFormControl-root-170 MuiFormControl-marginDense-172"
          style={
            Object {
              "marginRight": "5px",
            }
          }
        >
          <label
            className="MuiFormLabel-root-185 MuiInputLabel-root-174 MuiInputLabel-formControl-179 MuiInputLabel-animated-182 MuiInputLabel-marginDense-180 MuiInputLabel-outlined-184"
            data-shrink={false}
          >
            Name
          </label>
          <div
            className="MuiInputBase-root-205 MuiOutlinedInput-root-192 MuiInputBase-formControl-206 MuiInputBase-marginDense-212"
            onClick={[Function]}
          >
            <fieldset
              aria-hidden={true}
              className="MuiPrivateNotchedOutline-root-222 MuiOutlinedInput-notchedOutline-199"
              style={
                Object {
                  "paddingLeft": 8,
                }
              }
            >
              <legend
                className="MuiPrivateNotchedOutline-legend-223"
                style={
                  Object {
                    "width": 0.01,
                  }
                }
              >
                <span
                  dangerouslySetInnerHTML={
                    Object {
                      "__html": "&#8203;",
                    }
                  }
                />
              </legend>
            </fieldset>
            <input
              aria-invalid={false}
              className="MuiInputBase-input-215 MuiOutlinedInput-input-200 MuiInputBase-inputMarginDense-216 MuiOutlinedInput-inputMarginDense-201"
              disabled={false}
              name="additionalParameterName"
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              placeholder="Name"
              required={false}
              type="text"
              value=""
            />
          </div>
        </div>
        <div
          className="MuiFormControl-root-170 MuiFormControl-marginDense-172"
          style={
            Object {
              "marginRight": "5px",
            }
          }
        >
          <label
            className="MuiFormLabel-root-185 MuiInputLabel-root-174 MuiInputLabel-formControl-179 MuiInputLabel-animated-182 MuiInputLabel-marginDense-180 MuiInputLabel-outlined-184"
            data-shrink={false}
          >
            Value
          </label>
          <div
            className="MuiInputBase-root-205 MuiOutlinedInput-root-192 MuiInputBase-formControl-206 MuiInputBase-marginDense-212"
            onClick={[Function]}
          >
            <fieldset
              aria-hidden={true}
              className="MuiPrivateNotchedOutline-root-222 MuiOutlinedInput-notchedOutline-199"
              style={
                Object {
                  "paddingLeft": 8,
                }
              }
            >
              <legend
                className="MuiPrivateNotchedOutline-legend-223"
                style={
                  Object {
                    "width": 0.01,
                  }
                }
              >
                <span
                  dangerouslySetInnerHTML={
                    Object {
                      "__html": "&#8203;",
                    }
                  }
                />
              </legend>
            </fieldset>
            <input
              aria-invalid={false}
              className="MuiInputBase-input-215 MuiOutlinedInput-input-200 MuiInputBase-inputMarginDense-216 MuiOutlinedInput-inputMarginDense-201"
              disabled={false}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              placeholder="Value"
              required={false}
              type="text"
              value=""
            />
          </div>
        </div>
        <button
          className="MuiButtonBase-root-131 MuiButtonBase-disabled-132 MuiButton-root-240 MuiButton-contained-251 MuiButton-raised-254 MuiButton-sizeLarge-264 MuiButton-disabled-260 addAdditionalParameter"
          disabled={true}
          onBlur={[Function]}
          onClick={[Function]}
          onContextMenu={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          style={
            Object {
              "marginTop": "10px",
            }
          }
          tabIndex="-1"
          type="button"
        >
          <span
            className="MuiButton-label-241"
          >
            <svg
              aria-hidden="true"
              className="MuiSvgIcon-root-231"
              focusable="false"
              role="presentation"
              viewBox="0 0 24 24"
            >
              <path
                d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
              />
              <path
                d="M0 0h24v24H0z"
                fill="none"
              />
            </svg>
             Add parameter
          </span>
        </button>
      </div>
    </div>
  </form>
</div>
`;
